app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/email/CheckYourEmailController.scala

```diff
@@ -164,6 +169,14 @@ class CheckYourEmailController @Inject() (
               "[CheckYourEmailController][sendVerification] - " +
                 "Unable to send email verification request. Service responded with 'already verified'"
             )
+            if (service.enrolmentKey == Service.cds.enrolmentKey)
+              for {
+                maybeEori <- cdsFrontendDataCache.eori
+                verifiedEmailStatus <- Future.successful(
+                  updateVerifiedEmailService.updateVerifiedEmail(None, email, maybeEori.get)
```

Comment:
`maybeEori.get `will lead to ugly runtime errors, this should be replaced by matching on the underlying optional value or using `getOrElse`/`fold` to signal an exception.